### PR TITLE
issue #762 - added documentation about re-using the environment state

### PR DESCRIPTION
**Pull request by vavdoshka**

### CODE REVIEW

File: docs/writing-helmfile.md [hunk 1]
```diff
@@ -373,3 +373,42 @@ releases:
...
+  prod:
+    values:
+    - env/prod.yaml
+---
+
+helmfiles:
+- path: releases/myrelease/helmfile.yaml
+  values:
+  - {{ toYaml .Values | nindent 4 }}
+  # add other values sources if want to use overlay logic here
```
Comment (itscaro) [inline docs/writing-helmfile.md:397]:
```suggestion
  # pass the current state values to the sub-helmfile
  # add other values to use overlay logic here
```

File: docs/writing-helmfile.md [hunk 2]
```diff
@@ -373,3 +373,42 @@ releases:
...
+```
+
+and `releases/myrelease/helmfile.yaml` is as DRY as
+
+```yaml
+releases:
+- name: mychart-{{ .Values.myrelease.myname }}
+  installed: {{ .Values | get "myrelease.enabled" false }}
+
+  chart: mychart
```
Comment (itscaro) [inline docs/writing-helmfile.md:407]:
```suggestion
  installed: {{ .Values | get "myrelease.enabled" false }}
  chart: mychart
```

File: docs/writing-helmfile.md [hunk 3]
```diff
@@ -373,3 +373,42 @@ releases:
...
+  version: {{ .Values.myrelease.version }}
+  labels:
+    chart: mychart
+  values:
+    - values.yaml.gotmpl
+    # values templates would also inheret the values passed from upstream
```
Comment (itscaro) [inline docs/writing-helmfile.md:413]:
```suggestion
  values:
  - values.yaml.gotmpl
  # templated values would also inherit the values passed from upstream
```

**Comment by itscaro** [comment]
LGTM, thank you @vavdoshka

**Review by mumoshu — APPROVED** [review]
LGTM. Thank you so much for your contribution @vavdoshka ☺️

**Pull request merged**